spec/datadog/core/crashtracking/component_spec.rb

```diff
@@ -0,0 +1,351 @@
+          let(:access_log_buffer) { StringIO.new }
+          let(:access_log) { [[access_log_buffer, WEBrick::AccessLog::COMBINED_LOG_FORMAT]] }
+          let(:server_proc) do
+            proc do |req, res|
+              messages << req.tap { req.body } # Read body, store message before socket closes.
+              res.body = '{}'
+            end
+          end
+          let(:init_signal) { Queue.new }
+
+          let(:messages) { [] }
+
+          before do
+            server.mount_proc('/', &server_proc)
+            @server_thread = Thread.new { server.start }
+            init_signal.pop
+          end
+
+          after do
+            unless RSpec.current_example.skipped?
+              # When the test is skipped, server has not been initialized and @server_thread would be nil; thus we only
+              # want to touch them when the test actually run, otherwise we would cause the server to start (incorrectly)
+              # and join to be called on a nil @server_thread
+              server.shutdown
+              @server_thread.join
+            end
+          end
+        end
+
+        include_context 'HTTP server'
+
+        let(:request) { messages.first }
+        let(:port) { server[:Port] }
+
+        let(:agent_base_url) { "http://#{hostname}:#{port}" }
+
+        [:fiddle, :signal].each do |trigger|
+          it "reports crashes via http when app crashes with #{trigger}" do
+            fork_expectations = proc do |status:, stdout:, stderr:|
+              expect(Signal.signame(status.termsig)).to eq('SEGV').or eq('ABRT')
+              expect(stderr).to include('[BUG] Segmentation fault')
+            end
+
+            expect_in_fork(fork_expectations: fork_expectations) do
+              crash_tracker = build_crashtracker(agent_base_url: agent_base_url)
+              crash_tracker.start
+
+              if trigger == :fiddle
+                Fiddle.free(42)
+              else
+                Process.kill('SEGV', Process.pid)
+              end
+            end
+
+            crash_report = JSON.parse(request.body, symbolize_names: true)[:payload].first
+
+            expect(crash_report[:stack_trace]).to_not be_empty
+            expect(crash_report[:tags]).to include('signum:11', 'signame:SIGSEGV')
+
+            crash_report_message = JSON.parse(crash_report[:message], symbolize_names: true)
+
+            expect(crash_report_message[:metadata]).to include(
+              profiling_library_name: 'dd-trace-rb',
+              profiling_library_version: Datadog::VERSION::STRING,
+              family: 'ruby',
+              tags: ['tag1:value1', 'tag2:value2'],
+            )
+            expect(crash_report_message[:files][:'/proc/self/maps']).to_not be_empty
+            expect(crash_report_message[:os_info]).to_not be_empty
+          end
+        end
+
+        context 'when forked' do
+          it 'ensures the latest configuration applied' do
```

Comment:
Since this is a very non-trivial issue (e.g. it took us multiple minutes to discuss and understand it), I think this would benefit from a comment explaining why exactly this test is here and what's the subtle problem being tested.

It's true that we have test coverage for the issue now, and that's great because this test will fail if someone undoes the very subtle change we introduced to fix this problem.

BUT once this test starts failing, without a good comment, whoever makes it fail will need to "reverse engineer" why this test is here and what's it doing. (They may also arrive to the conclusion that the test doesn't make any sense and choose to remove it instead, which would be doubly bad)